File: JUNGOL/Beginner/#1402 약수 구하기/약수 구하기.cpp
#include <iostream>

using namespace std;

int main() {
	int n,k; cin>>n>>k;

	if(k == 1){
		printf("1\n");
		return 0;
	}

	int cnt = 1;
	for(int i=2; i<=n; i++){
		if(!(n%i)) cnt++;
		if(cnt == k){
			printf("%d\n", i);
			return 0;
		}
	}

	printf("0\n");
}

File: JUNGOL/Beginner/#1402 약수 구하기/약수 구하기.py
def main():
    n,k = map(int, input().split())

    if k == 1:
        print(1)
        return
    
    cnt = 1
    for i in range(2, n+1):
        if n%i == 0: cnt += 1
        if cnt == k:
            print(i)
            return
        
    print(0)

main()
